tests/test_quote.py

```diff
@@ -0,0 +1,461 @@
+    quote_str,
+    quotes_query,
+    update_quote,
+)
+from config import CONFIG
+from models import Base, Quote, QuoteOptouts, User
+from utils.mentions import parse_mention
+
+TEST_QUOTES = [
+    Quote(
+        author_type="id",
+        author_id=1,
+        author_string=None,
+        quote="talking to myself!",
+        created_at=datetime(2018, 10, 11),
+        edited=False,
+        edited_at=None,
+    ),
+    Quote(
+        author_type="string",
+        author_id=None,
+        author_string="ircguy",
+        quote="talking to myself! on irc!",
+        created_at=datetime(2018, 10, 12),
+        edited=False,
+        edited_at=None,
+    ),
+    Quote(
+        author_type="id",
+        author_id=2,
+        author_string=None,
+        quote="talking to someone else!",
+        created_at=datetime(2018, 10, 13),
+        edited=False,
+        edited_at=None,
+    ),
+    Quote(
+        author_type="string",
+        author_id=None,
+        author_string="ircguy",
+        quote="talking to someone else! on irc!",
+        created_at=datetime(2018, 10, 14),
+        edited=False,
+        edited_at=None,
+    ),
+    Quote(
+        author_type="id",
+        author_id=1,
+        author_string=None,
+        quote="talking about someone else! from irc!",
+        created_at=datetime(2018, 10, 15),
+        edited=False,
+        edited_at=None,
+    ),
+    Quote(
+        author_type="string",
+        author_id=None,
+        author_string="ircguy2",
+        quote="something about FOSS idk",
+        created_at=datetime(2018, 10, 16),
+        edited=False,
+        edited_at=None,
+    ),
+]
+
+
+@pytest.fixture(scope="module")
+def database():
+    # Locate the testing config for Alembic
+    config = Config(os.path.join(os.path.dirname(__file__), "../alembic.tests.ini"))
+
+    # Set the migration secret key here
+    if not os.environ.get("SECRET_KEY", None):
+        os.environ["SECRET_KEY"] = "test"
+
+    # Start up the in-memory database instance
+    db_engine = create_engine("sqlite:///:memory:")
+    Base.metadata.create_all(db_engine)
+    db_session = Session(bind=db_engine)
+
+    # Mark it as up-to-date with migrations
+    command.stamp(config, "head")
+
+    # Add three users
+    users = [
+        User(user_uid=1000, username="Foo"),
+        User(user_uid=1337, username="Bar"),
+        User(user_uid=3000, username="NoQuoteMePls"),
+    ]
+
+    db_session.add_all(users)
+
+    db_session.add_all(TEST_QUOTES)
+
+    # Add some opt-outs
+    db_session.add(QuoteOptouts(user_type="id", user_id=3, user_string=None))
+
+    db_session.commit()
+    return db_session
+
+
+QUERY_QUOTES = {
+    "By ID": ("#1", [TEST_QUOTES[0]]),
+    "By Author (Discord)": ("<@!1337>", [TEST_QUOTES[2]]),
+    "By Author (IRC/String)": ("@ircguy", [TEST_QUOTES[1], TEST_QUOTES[3]]),
+    "By Topic": ("irc", [TEST_QUOTES[1], TEST_QUOTES[3], TEST_QUOTES[4]]),
+    "No valid quotes": ("blargleargle", []),
+}
+
+ADD_QUOTES = {
```

Comment:
Not sure I like testing the strings returned by the bot. I'd consider creating enums which encapsulate the possible return states and letting the command itself decide how to turn those into a string.